tasks/sync-granule/index.js

```diff
@@ -9,16 +11,61 @@ const { s3Join } = require('@cumulus/aws-client/S3');
 const log = require('@cumulus/common/log');
 const GranuleFetcher = require('./GranuleFetcher');
 
+/**
```

Comment:
I think it would be easier to parse this is ProviderConfig and CollectionConfig etc (or some such names) were broken out and defined instead of all being in this one massive definition

Reply:
It's a good point - this was a direct transcription of the config object, but thinking about this more..... this is a duplicate of the API type.  

If we had more work in this space I'm starting to think that we'd be better served with .ts definitions rather than json schemas.

Reply:
@etcart  see what you think re: 6582d7ec265 3e3deff011e

Reply:
6582d7ec265 3e3deff011e

Reply:
yea, I appreciate the update I find this much more understandable

Reply:
and yea, proper typing would be great here